### ISSUE TITLE
Unable to launch kernel "MIOPEN_DEBUG_CONV_DIRECT_ASM_5X10U2V2"

### ISSUE THREAD

**Issue by asroy** [ROCmSoftwarePlatform/MIOpen#895]
See detailed issue found in JIRA http://ontrack-internal.amd.com/browse/SWDEV-277901
```
MIOpenDriver conv -W 341 -H 79 -c 32 -n 4 -k 32 -y 5 -x 10 -p 0 -q 0 -u 2 -v 2 -t 1 -V 0 -i 1
MIOpen Error: /root/Staging/MathLibs/MLOpen/src/hipoc/hipoc_kernel.cpp:86: Failed to launch kernel hipErrorLaunchFailure
RunForwardGPU() failed, rc = 0x7
```
I've narrow it done to a ASM kernel, by disabling it, the issue is gone 
```
export MIOPEN_DEBUG_CONV_DIRECT_ASM_5X10U2V2=0
```

The original issue is suspecting the root cause is that the number of work-item per work-group specified in the kernel is different from the actual number when launch the kernel.
```
if the actual launch param exceed the source specified launch param then the runtime will reject the launch
```

However, I'm not able to find anything wrong on this aspect, so I don't know what the root cause of the failure.

Step to reproduce the failure:
1. launch the docker as given in the JIRA
2. cd /root/driver/MLOpen/build_hip/
3. make -j MIOpenDriver
4. export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/opt/rocm-4.2.0-6608/lib
5. bin/MIOpenDriver conv -W 341 -H 79 -c 32 -n 4 -k 32 -y 5 -x 10 -p 0 -q 0 -u 2 -v 2 -t 1 -V 0 -i 1

**Comment by asroy** [comment]
@muralinr Please look into this issue to find out if this is a launch parameter issue or not. Need you assembly expertise on this issue

cc @JehandadKhan @atamazov

**Comment by muralinr** [comment]
Hello, It's not a MIOPEN issue. I looked at ASM source code and kernel launch solver code to cross check it.  Workgroup info  specified is same in both files.

ASM source filev (kernels/conv5x10u2v2f1.s)  
.reqd_workgroup_size: [ 64, 8, 1 ]

Solver file (solver/conv_asm_5x10u2v2f1.cpp):
construction_params.l_wk.push_back(64);
construction_params.l_wk.push_back(8);
construction_params.l_wk.push_back(1);

**Comment by atamazov** [comment]
Should be fixed by #931.
